R/assertions.R

```diff
@@ -0,0 +1,82 @@
+#' @export
+validate <- function(data, name, description = NULL) {
+  if (missing(name)) {
+    name <- deparse(substitute(data))
+    if (name == ".") {
+      name <- get_first_name(data)
+    }
+  }
+  attr(data, "data-description") <- description
+  attr(data, "data-name") <- name
+  attr(data, "assertr_in_chain_success_fun_override") <- assertr::success_append
+  attr(data, "assertr_in_chain_error_fun_override") <- assertr::error_append
+  data
+}
+
+#' @export
+assert_if <- function(data, expr, description = NA, obligatory = FALSE) {
+  assertr::verify(data = data,
+                  expr = !!rlang::enexpr(expr),
+                  description = format_description(description),
+                  skip_chain_opts = FALSE,
+                  obligatory = obligatory,
+                  success_fun = assertr::success_append,
+                  error_fun = assertr::error_append,
+                  defect_fun = assertr::defect_append)
+}
+
+#' @export
+assert_cols <- function(data, predicate, ..., predicate_calculated_from_column = FALSE, obligatory = FALSE, description = NA) {
```

Comment:
Thank you so much for pushing about the naming. I just realised that the problem is because of ellipsis (...) used for passing column names to validate. This forces you to read the documentation and verify what is this ellipsis for, but moreover it complicates named/unnamed argument usage. When I removed it, I also realised that is causes funny code duplication (`assert` and `insist` are separate functions only because there is an ellipsis and we needed separate arguments `predicate` and `predicate_generator` that couldn't be used together:

These functions are in fact the same:

```
insist(data, predicate_generator, ..., success_fun = success_continue,
       error_fun = error_stop, skip_chain_opts = FALSE,
       obligatory = FALSE, defect_fun = defect_append, description = NA)

assert(data, predicate, ..., success_fun = success_continue,
       error_fun = error_stop, skip_chain_opts = FALSE,
       obligatory = FALSE, defect_fun = defect_append, description = NA)
```

Please take a look if you agree with my new approach.

Reply:
Yes I agree but unfortunately we cannot replace ellipsis with `cols` argument. With such approach we're unable to pass multiple columns to the function what is the standard approach.
I'd suggest to:
1. Revert to using the ellipsis.
2. Validate if we don't pass names arguments other than the one offered by insist and assert.
I'll take care of it.

Reply:
In the ideal world we could completely skip using separate predicate and predicate_generator arguments.
We'll just have to detect which predicate is doing what and then apply proper function. Let's stay with two parameters now but I'll try to propse sth soon.